sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -176,6 +189,32 @@ public String getDescription() {
     }
   }
 
+  @Immutable
+  private enum FollowParentSampler implements Sampler {
+    INSTANCE;
+
+    // If a parent is set, always follows the same sampling decision as the parent.
+    // Otherwise, always makes a "yes" decision on {@link Span} sampling.
```

Comment:
It seems like follow parent should be a setting on any sampler, or we should support multiple samplers where any sampler can return something like `NO_DECISION` to allow another sampler to pick it up. Always yes seems to be unrelated to following the parent, right?

Reply:
Oops just jumped through bogdan's link, yeah shouldn't we use parentorelse instead of adding this sampler?

I would argue ParentOrElse(AlwaysOn) should be the default for the SDK though

Reply:
@anuraaga can we argue in a separate PR :)